utils/src/shutdown.rs

```diff
@@ -0,0 +1,163 @@
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use tokio::sync::watch;
+use tracing::warn;
+
+/// Shutdown Signal
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
```

Comment:
I think we don't need to derive `PartialEq` and `Eq` here.

Reply:
Fixed